Add medals

SuwannaTruss opened the pull request:
https://trello.com/c/m34Fg61p/59-add-images-for-the-medals-and-a-function-to-select-the-correct-medal-based-on-number-of-confirmed-views-made-by-the-user

Pending
- to find different images foe each rank

SuwannaTruss committed:
added getCurrentUserSightingsCount endpoint

SuwannaTruss committed:
connect to frontend and only count confirmed report sightings

oskarwilliams committed:
Move get user sightings to user controller and add to user api model

oskarwilliams commented on WhaleSpotting/react-app/src/components/Profile.tsx:
```diff
@@ -57,6 +83,7 @@ export function Profile(): JSX.Element {
             fetchCurrentUserSightings(page)
                 .then(data => setData(data));
         }
+        assignRank();
```
Can we call this after we call getUser() instead, as in do getUser().then(() => assignRank()) on line24.

SuwannaTruss replied:
done.

SuwannaTruss replied:
We found a problem that the page re-render which cause the Rank to somehow reset to newbie. 
I have temporary commented out and replace it with getRankSrc function.
But it now displays the correct badges.

oskarwilliams commented on WhaleSpotting/react-app/src/components/Profile.tsx:
```diff
@@ -105,7 +132,7 @@ export function Profile(): JSX.Element {
                 <h2 className="heading">Your {feedToggle}</h2>
                 <div className="card-holder">
-                    {cards.length === 0 && page === 1 && feedToggle === "Sightings" ? <div className="card-component">Nothing here, <Link to="reportsighting"> report a sighting </ Link> </div> : cards}
+                    {cards.length === 0 && page === 1 && feedToggle === "" ? <div className="card-component">Nothing here, <Link to="reportsighting"> report a sighting </ Link> </div> : cards}
```
So this wont show anything for Sightings?

SuwannaTruss replied:
Only when there are no sightings submitted by the user, it will show the link to direct them to Report sighting (on the sighting feed), so I think we need feedToggle === "Sightings". And I think it works before.

SuwannaTruss replied:
![image](https://user-images.githubusercontent.com/67764574/129219653-e341a136-49f0-46e9-92a0-be078770e2bf.png)

SuwannaTruss replied:
with feedToggle === "", it look like this.
![image](https://user-images.githubusercontent.com/67764574/129219928-be9627f3-84c5-4ad6-83f2-3726761b6a17.png)

SuwannaTruss committed:
added badge images

SuwannaTruss committed:
updated photos

SuwannaTruss committed:
updated rank condition - WIP

SuwannaTruss committed:
Temporary solution to display correct batch

- the assignRank seems  to be re-rendered and the badge were changed from the correct badge to Newbie badge.

SuwannaTruss committed:
Merge remote-tracking branch 'origin/main' into AddMedals

oskarwilliams committed:
Minor tidying
